client/matching/client.go

```diff
@@ -317,9 +333,39 @@ func (c *clientImpl) RefreshTaskListPartitionConfig(
 	request *types.MatchingRefreshTaskListPartitionConfigRequest,
 	opts ...yarpc.CallOption,
 ) (*types.MatchingRefreshTaskListPartitionConfigResponse, error) {
-	peer, err := c.peerResolver.FromTaskList(request.TaskList.GetName())
+	peer, err := c.getShardOwner(ctx, request.TaskList.GetName())
 	if err != nil {
 		return nil, err
 	}
 	return c.client.RefreshTaskListPartitionConfig(ctx, request, append(opts, yarpc.WithShardKey(peer))...)
 }
+
+func (c *clientImpl) getShardOwner(ctx context.Context, taskListName string) (string, error) {
```

Comment:
We have the peer resolver interface. 
Could you consider implementing a new type of peer resolver and put the logic in the new implementation?

Reply:
+1. Maybe introduce a MultiPeerResolver object which satisfies PeerResolver interface. It can do take this responsibility (either calls legacy peerResolver or shardDistributor)

Reply:
Good points, will work on making this. Thanks!